com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/service/CopybookServiceE2ETest.java

```diff
@@ -79,62 +84,60 @@ public void initActivities() {
    * file.
    */
   @Test
-  public void generateDependencyFileOnCallbackPositiveTest() {
-
-    // generate a required copybook event
+  public void publishRequestOfCopybook_createDependencyFile() {
     broker.postData(
         RequiredCopybookEvent.builder()
             .name(CPY_NAME_WITHOUT_EXT)
-            .documentUri(DOCUMENT_URI)
+            .documentUri(COBOL_FILE_DOCUMENT_URI)
             .textDocumentSyncType("DID_OPEN")
             .build());
+
     // after one second is expected to found the dep file on filesystem
-    waitAndAssert(Boolean.TRUE);
+    waitAndAssert_DepFileIsCreated(Boolean.TRUE);
   }
 
   /**
    * This test verify that when the preprocessor publish a require copybook event without specifying
    * the TextDocumentSync type, the FileSystem service will not write/update any dependency file.
    */
   @Test
-  public void NotGenerateDependencyFileOnCallbackNegativeTest() {
-    // generate a required copybook event
+  public void noTextDocSyncTypeDefined_NoDependencyFileIsCreated() {
     broker.postData(
         RequiredCopybookEvent.builder()
             .name(CPY_NAME_WITHOUT_EXT)
-            .documentUri(DOCUMENT_URI)
+            .documentUri(COBOL_FILE_DOCUMENT_URI)
             .build());
+
     // after one second is expected to found the dep file on filesystem
-    waitAndAssert(Boolean.FALSE);
+    waitAndAssert_DepFileIsCreated(Boolean.FALSE);
   }
 
   /**
    * This test verify that when the preprocessor publish a require copybook event when the document
    * is in DID_CHANGE mode and the FileSystem service will not write/update any dependency file.
    */
   @Test
-  public void NotGenerateDependencyFileOnDidChangeTest() {
-    // generate a required copybook event
+  public void DocumentInDidChangeMode_NoDepFileIsCreated() {
     broker.postData(
         RequiredCopybookEvent.builder()
             .name(CPY_NAME_WITHOUT_EXT)
-            .documentUri(DOCUMENT_URI)
+            .documentUri(COBOL_FILE_DOCUMENT_URI)
             .textDocumentSyncType("DID_CHANGE")
             .build());
     // after one second is expected to found the dep file on filesystem
-    waitAndAssert(Boolean.FALSE);
+    waitAndAssert_DepFileIsCreated(Boolean.FALSE);
   }
 
   /**
    * This test verify that when a non required event is published on the databus, the dependency
    * file is not created. This test assume that .cobdep folder exists but no depfile is created.
    */
   @Test
-  public void noDependencyFileWithOtherEvent() {
+  public void unknownEventPublished_NoDepFileIsCreated() {
```

Comment:
I think we will receive a method name code smell here and for two more methods in this class

Reply:
Purpose of unit test is to be clear for who run the test. There is a code smell about regex for the method (underline is not allowed). But for me help the developer to understand the purpose of the test is much more important then the camel case that in past created cryptic method name.
And I applied this practices only on unit test methods.